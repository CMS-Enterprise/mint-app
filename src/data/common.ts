// eslint-disable-next-line import/prefer-default-export
export const yesNoMap: any = {
  YES: 'Yes',
  NO: 'No',
  NOT_SURE: "I'm not sure"
};
