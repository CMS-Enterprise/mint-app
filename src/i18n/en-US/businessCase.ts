const businessCase = {
  submission: {
    confirmation: {
      heading: 'Your Business Case has been submitted',
      subheading: 'Your reference ID is {{referenceId}}',
      homeCta: 'Go back to EASi homepage',
      taskListCta: 'Go back to Governance Task List'
    }
  },
  grtFeedback: {
    header: 'Recommendations',
    grbSubhead: 'GRT recommendations to the GRB',
    dateSRHelpText: 'Feedback from the meeting on {{date}}',
    grbHelpText:
      'These are the Governance Review Team recommendations for the Governance Review Board.',
    businessOwnerSubhead: 'GRT recommendations to the Business Owner',
    businessOwnerHelpText:
      'These are the Governance Review Team recommendations for the Business Owner.'
  }
};

export default businessCase;
