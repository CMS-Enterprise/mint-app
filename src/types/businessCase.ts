import { DateTime } from 'luxon';

import { LifecycleCosts } from 'types/estimatedLifecycle';
import { SystemIntakeStatus } from 'types/systemIntake';

export type EstimatedLifecycleCostLines = {
  year1: LifecycleCosts;
  year2: LifecycleCosts;
  year3: LifecycleCosts;
  year4: LifecycleCosts;
  year5: LifecycleCosts;
};

// Base Solution
export type BusinessCaseSolution = {
  title: string;
  summary: string;
  pros: string;
  cons: string;
  estimatedLifecycleCost: EstimatedLifecycleCostLines;
  costSavings: string;
};

// Proposed Solution
export type ProposedBusinessCaseSolution = BusinessCaseSolution & {
  acquisitionApproach: string;
  security: {
    isApproved: boolean | null;
    isBeingReviewed: string;
  };
  hosting: {
    type: string;
    location: string;
    cloudServiceType?: string;
  };
  hasUserInterface: string;
};

type BusinessCaseStatus = 'OPEN' | 'CLOSED';

export type GeneralRequestInfoForm = {
  requestName: string;
  requester: {
    name: string;
    phoneNumber: string;
  };
  businessOwner: {
    name: string;
  };
};

export type RequestDescriptionForm = {
  businessNeed: string;
  cmsBenefit: string;
  priorityAlignment: string;
  successIndicators: string;
};

export type AsIsSolutionForm = {
  asIsSolution: BusinessCaseSolution;
};

export type PreferredSolutionForm = {
  preferredSolution: ProposedBusinessCaseSolution;
};

export type AlternativeASolutionForm = {
  alternativeA: ProposedBusinessCaseSolution;
};

export type AlternativeBSolutionForm = {
  alternativeB: ProposedBusinessCaseSolution;
};

// Business Case Form Model
export type BusinessCaseModel = GeneralRequestInfoForm &
  RequestDescriptionForm &
  AsIsSolutionForm &
  PreferredSolutionForm &
  AlternativeASolutionForm &
  AlternativeBSolutionForm & {
    id?: string;
    euaUserId?: string;
    status: BusinessCaseStatus;
    systemIntakeId: string;
    systemIntakeStatus: SystemIntakeStatus;
    initialSubmittedAt?: string;
    lastSubmittedAt?: string;
    createdAt: string;
  };

export type BusinessCasesState = {
  businessCases: BusinessCaseModel[];
  isLoading: boolean | null;
  loadedTimestamp: DateTime | null;
  error: string | null;
};

// Redux store type for business case
export type BusinessCaseState = {
  form: BusinessCaseModel;
  isLoading: boolean | null;
  isSaving: boolean;
  error: any;
};
