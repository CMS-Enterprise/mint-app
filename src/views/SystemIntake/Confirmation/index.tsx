import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useParams } from 'react-router-dom';
import { IconNavigateBefore } from '@trussworks/react-uswds';

import PageHeading from 'components/PageHeading';
import { ImproveEasiSurvey } from 'components/Survey';

const Confirmation = () => {
  const { systemId } = useParams<{ systemId: string }>();
  const { t } = useTranslation('intake');

  return (
    <div className="grid-container margin-bottom-7">
      <div>
        <PageHeading>{t('submission.confirmation.heading')}</PageHeading>
        <h2 className="margin-bottom-8 text-normal">
          {t('submission.confirmation.subheading', {
            referenceId: systemId
          })}
        </h2>
        <ImproveEasiSurvey />
        <div>
          <Link
            to={`/governance-task-list/${systemId}`}
            className="display-flex"
          >
            <IconNavigateBefore className="margin-x-05" aria-hidden />
            {t('submission.confirmation.taskListCta')}
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Confirmation;
