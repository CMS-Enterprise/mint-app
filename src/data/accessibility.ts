// Data for the Accessibility/508 process
import { AccessibilityRequestForm } from 'types/accessibility';

// eslint-disable-next-line
export const initialAccessibilityRequestFormData: AccessibilityRequestForm = {
  intakeId: '',
  businessOwner: {
    name: '',
    component: ''
  },
  requestName: ''
};
