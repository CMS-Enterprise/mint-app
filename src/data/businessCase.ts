import cloneDeep from 'lodash/cloneDeep';

import {
  BusinessCaseModel,
  EstimatedLifecycleCostLines,
  ProposedBusinessCaseSolution
} from 'types/businessCase';
import { LifecycleCosts } from 'types/estimatedLifecycle';

const emptyPhaseValues = {
  development: {
    isPresent: false,
    cost: ''
  },
  operationsMaintenance: {
    isPresent: false,
    cost: ''
  },
  other: {
    isPresent: false,
    cost: ''
  }
};

export const defaultEstimatedLifecycle = {
  year1: cloneDeep(emptyPhaseValues),
  year2: cloneDeep(emptyPhaseValues),
  year3: cloneDeep(emptyPhaseValues),
  year4: cloneDeep(emptyPhaseValues),
  year5: cloneDeep(emptyPhaseValues)
};

export const defaultProposedSolution = {
  title: '',
  summary: '',
  acquisitionApproach: '',
  pros: '',
  cons: '',
  estimatedLifecycleCost: cloneDeep(defaultEstimatedLifecycle),
  costSavings: '',
  security: {
    isApproved: null,
    isBeingReviewed: ''
  },
  hosting: {
    type: '',
    location: '',
    cloudServiceType: ''
  },
  hasUserInterface: ''
};

export const businessCaseInitialData: BusinessCaseModel = {
  status: 'OPEN',
  systemIntakeId: '',
  systemIntakeStatus: '',
  requestName: '',
  requester: {
    name: '',
    phoneNumber: ''
  },
  businessOwner: {
    name: ''
  },
  businessNeed: '',
  cmsBenefit: '',
  priorityAlignment: '',
  successIndicators: '',
  asIsSolution: {
    title: '',
    summary: '',
    pros: '',
    cons: '',
    estimatedLifecycleCost: cloneDeep(defaultEstimatedLifecycle),
    costSavings: ''
  },
  preferredSolution: cloneDeep(defaultProposedSolution),
  alternativeA: cloneDeep(defaultProposedSolution),
  alternativeB: cloneDeep(defaultProposedSolution),
  createdAt: ''
};

type lifecycleCostLinesType = {
  'As Is': EstimatedLifecycleCostLines;
  Preferred: EstimatedLifecycleCostLines;
  A: EstimatedLifecycleCostLines;
  B: EstimatedLifecycleCostLines;
};

/**
 * This function tells us whether the parameter alternativeSolution has been started
 * @param alternativeSolution - an alternative solution case (e.g. A, B)
 */
export const alternativeSolutionHasFilledFields = (
  alternativeSolution: ProposedBusinessCaseSolution
) => {
  if (!alternativeSolution) {
    return false;
  }

  const {
    title,
    summary,
    acquisitionApproach,
    security,
    hosting,
    hasUserInterface,
    pros,
    cons,
    costSavings,
    estimatedLifecycleCost
  } = alternativeSolution;

  let hasLineItem;
  Object.values(estimatedLifecycleCost).forEach(phase => {
    if (
      phase.development.isPresent ||
      phase.operationsMaintenance.isPresent ||
      phase.other.isPresent
    ) {
      hasLineItem = true;
    }
  });

  return (
    title ||
    summary ||
    acquisitionApproach ||
    security.isApproved ||
    security.isBeingReviewed ||
    hosting.type ||
    hosting.location ||
    hosting.cloudServiceType ||
    hasUserInterface ||
    pros ||
    cons ||
    costSavings ||
    hasLineItem
  );
};

export const prepareBusinessCaseForApp = (
  businessCase: any
): BusinessCaseModel => {
  const phaseTypeMap: any = {
    Development: 'development',
    'Operations and Maintenance': 'operationsMaintenance',
    Other: 'other'
  };

  const lifecycleCostLines: lifecycleCostLinesType = {
    'As Is': cloneDeep(defaultEstimatedLifecycle),
    Preferred: cloneDeep(defaultEstimatedLifecycle),
    A: cloneDeep(defaultEstimatedLifecycle),
    B: cloneDeep(defaultEstimatedLifecycle)
  };

  let doesAltBHaveLifecycleCostLines = false;

  businessCase.lifecycleCostLines.forEach((line: any) => {
    const phaseType: 'development' | 'operationsMaintenance' | 'other' =
      phaseTypeMap[`${line.phase}`];

    if (line.solution === 'B') {
      doesAltBHaveLifecycleCostLines = true;
    }

    lifecycleCostLines[line.solution as keyof lifecycleCostLinesType][
      `year${line.year}` as keyof EstimatedLifecycleCostLines
    ][phaseType] = {
      isPresent: !!line.cost,
      cost: line.cost ? line.cost.toString() : ''
    };
  });

  if (!doesAltBHaveLifecycleCostLines) {
    lifecycleCostLines.B = cloneDeep(defaultEstimatedLifecycle);
  }

  return {
    id: businessCase.id,
    euaUserId: businessCase.euaUserId,
    status: businessCase.status,
    systemIntakeId: businessCase.systemIntakeId,
    systemIntakeStatus: businessCase.systemIntakeStatus,
    requestName: businessCase.projectName,
    requester: {
      name: businessCase.requester,
      phoneNumber: businessCase.requesterPhoneNumber
    },
    businessOwner: {
      name: businessCase.businessOwner
    },
    businessNeed: businessCase.businessNeed,
    cmsBenefit: businessCase.cmsBenefit,
    priorityAlignment: businessCase.priorityAlignment,
    successIndicators: businessCase.successIndicators,
    asIsSolution: {
      title: businessCase.asIsTitle,
      summary: businessCase.asIsSummary,
      pros: businessCase.asIsPros,
      cons: businessCase.asIsCons,
      costSavings: businessCase.asIsCostSavings,
      estimatedLifecycleCost: lifecycleCostLines['As Is']
    },
    preferredSolution: {
      title: businessCase.preferredTitle,
      summary: businessCase.preferredSummary,
      acquisitionApproach: businessCase.preferredAcquisitionApproach,
      pros: businessCase.preferredPros,
      cons: businessCase.preferredCons,
      costSavings: businessCase.preferredCostSavings,
      estimatedLifecycleCost: lifecycleCostLines.Preferred,
      security: {
        isApproved: businessCase.preferredSecurityIsApproved,
        isBeingReviewed: businessCase.preferredSecurityIsBeingReviewed
      },
      hosting: {
        type: businessCase.preferredHostingType,
        location: businessCase.preferredHostingLocation,
        cloudServiceType: businessCase.preferredHostingCloudServiceType
      },
      hasUserInterface: businessCase.preferredHasUI
    },
    alternativeA: {
      title: businessCase.alternativeATitle,
      summary: businessCase.alternativeASummary,
      acquisitionApproach: businessCase.alternativeAAcquisitionApproach,
      pros: businessCase.alternativeAPros,
      cons: businessCase.alternativeACons,
      costSavings: businessCase.alternativeACostSavings,
      estimatedLifecycleCost: lifecycleCostLines.A,
      security: {
        isApproved: businessCase.alternativeASecurityIsApproved,
        isBeingReviewed: businessCase.alternativeASecurityIsBeingReviewed
      },
      hosting: {
        type: businessCase.alternativeAHostingType,
        location: businessCase.alternativeAHostingLocation,
        cloudServiceType: businessCase.alternativeAHostingCloudServiceType
      },
      hasUserInterface: businessCase.alternativeAHasUI
    },
    alternativeB: {
      title: businessCase.alternativeBTitle || '',
      summary: businessCase.alternativeBSummary || '',
      acquisitionApproach: businessCase.alternativeBAcquisitionApproach || '',
      pros: businessCase.alternativeBPros || '',
      cons: businessCase.alternativeBCons || '',
      costSavings: businessCase.alternativeBCostSavings || '',
      estimatedLifecycleCost: lifecycleCostLines.B,
      security: {
        isApproved: businessCase.alternativeBSecurityIsApproved,
        isBeingReviewed: businessCase.alternativeBSecurityIsBeingReviewed || ''
      },
      hosting: {
        type: businessCase.alternativeBHostingType || '',
        location: businessCase.alternativeBHostingLocation || '',
        cloudServiceType: businessCase.alternativeBHostingCloudServiceType || ''
      },
      hasUserInterface: businessCase.alternativeBHasUI || ''
    },
    createdAt: businessCase.createdAt,
    initialSubmittedAt: businessCase.initialSubmittedAt,
    lastSubmittedAt: businessCase.lastSubmittedAt
  };
};

export const prepareBusinessCaseForApi = (
  businessCase: BusinessCaseModel
): any => {
  const alternativeBExists = alternativeSolutionHasFilledFields(
    businessCase.alternativeB
  );
  const solutionNameMap: {
    solutionLifecycleCostLines: EstimatedLifecycleCostLines;
    solutionApiName: string;
  }[] = [
    {
      solutionLifecycleCostLines:
        businessCase.asIsSolution.estimatedLifecycleCost,
      solutionApiName: 'As Is'
    },
    {
      solutionLifecycleCostLines:
        businessCase.preferredSolution.estimatedLifecycleCost,
      solutionApiName: 'Preferred'
    },
    {
      solutionLifecycleCostLines:
        businessCase.alternativeA.estimatedLifecycleCost,
      solutionApiName: 'A'
    },
    ...(alternativeBExists
      ? [
          {
            solutionLifecycleCostLines:
              businessCase.alternativeB.estimatedLifecycleCost,
            solutionApiName: 'B'
          }
        ]
      : [])
  ];

  const yearMap = (
    lifecycleCostLines: EstimatedLifecycleCostLines
  ): { phases: LifecycleCosts; year: string }[] => {
    return [
      { phases: lifecycleCostLines.year1, year: '1' },
      { phases: lifecycleCostLines.year2, year: '2' },
      { phases: lifecycleCostLines.year3, year: '3' },
      { phases: lifecycleCostLines.year4, year: '4' },
      { phases: lifecycleCostLines.year5, year: '5' }
    ];
  };

  const lifecycleCostLines = solutionNameMap
    .map(({ solutionLifecycleCostLines, solutionApiName }) => {
      return yearMap(solutionLifecycleCostLines)
        .map(({ phases, year }) => {
          const { development, operationsMaintenance, other } = phases;
          const developmentCost = {
            solution: solutionApiName,
            phase: 'Development',
            cost: development.isPresent ? parseFloat(development.cost) : null,
            year
          };
          const omCost = {
            solution: solutionApiName,
            phase: 'Operations and Maintenance',
            cost: operationsMaintenance.isPresent
              ? parseFloat(phases.operationsMaintenance.cost)
              : null,
            year
          };
          const otherCost = {
            solution: solutionApiName,
            phase: 'Other',
            cost: other.isPresent ? parseFloat(phases.other.cost) : null,
            year
          };
          return [developmentCost, omCost, otherCost];
        })
        .flat();
    })
    .flat();

  return {
    ...(businessCase.id && {
      id: businessCase.id
    }),
    ...(businessCase.euaUserId && {
      euaUserId: businessCase.euaUserId
    }),
    status: businessCase.status,
    systemIntakeId: businessCase.systemIntakeId,
    projectName: businessCase.requestName,
    requester: businessCase.requester.name,
    requesterPhoneNumber: businessCase.requester.phoneNumber,
    businessOwner: businessCase.businessOwner.name,
    businessNeed: businessCase.businessNeed,
    cmsBenefit: businessCase.cmsBenefit,
    priorityAlignment: businessCase.priorityAlignment,
    successIndicators: businessCase.successIndicators,
    asIsTitle: businessCase.asIsSolution.title,
    asIsSummary: businessCase.asIsSolution.summary,
    asIsPros: businessCase.asIsSolution.pros,
    asIsCons: businessCase.asIsSolution.cons,
    asIsCostSavings: businessCase.asIsSolution.costSavings,
    preferredTitle: businessCase.preferredSolution.title,
    preferredSummary: businessCase.preferredSolution.summary,
    preferredAcquisitionApproach:
      businessCase.preferredSolution.acquisitionApproach,
    preferredSecurityIsApproved:
      businessCase.preferredSolution.security.isApproved,
    preferredSecurityisBeingReviewed:
      businessCase.preferredSolution.security.isBeingReviewed,
    preferredHostingType: businessCase.preferredSolution.hosting.type,
    preferredHostingLocation: businessCase.preferredSolution.hosting.location,
    preferredHostingCloudServiceType:
      businessCase.preferredSolution.hosting.cloudServiceType,
    preferredHasUI: businessCase.preferredSolution.hasUserInterface,
    preferredPros: businessCase.preferredSolution.pros,
    preferredCons: businessCase.preferredSolution.cons,
    preferredCostSavings: businessCase.preferredSolution.costSavings,
    alternativeATitle: businessCase.alternativeA.title,
    alternativeASummary: businessCase.alternativeA.summary,
    alternativeAAcquisitionApproach:
      businessCase.alternativeA.acquisitionApproach,
    alternativeASecurityIsApproved:
      businessCase.alternativeA.security.isApproved,
    alternativeASecurityisBeingReviewed:
      businessCase.alternativeA.security.isBeingReviewed,
    alternativeAHostingType: businessCase.alternativeA.hosting.type,
    alternativeAHostingLocation: businessCase.alternativeA.hosting.location,
    alternativeAHostingCloudServiceType:
      businessCase.alternativeA.hosting.cloudServiceType,
    alternativeAHasUI: businessCase.alternativeA.hasUserInterface,
    alternativeAPros: businessCase.alternativeA.pros,
    alternativeACons: businessCase.alternativeA.cons,
    alternativeACostSavings: businessCase.alternativeA.costSavings,
    alternativeBTitle: alternativeBExists
      ? businessCase.alternativeB.title
      : null,
    alternativeBSummary: alternativeBExists
      ? businessCase.alternativeB.summary
      : null,
    alternativeBAcquisitionApproach: alternativeBExists
      ? businessCase.alternativeB.acquisitionApproach
      : null,
    alternativeBSecurityIsApproved: alternativeBExists
      ? businessCase.alternativeB.security.isApproved
      : null,
    alternativeBSecurityisBeingReviewed: alternativeBExists
      ? businessCase.alternativeB.security.isBeingReviewed
      : null,
    alternativeBHostingType: alternativeBExists
      ? businessCase.alternativeB.hosting.type
      : null,
    alternativeBHostingLocation: alternativeBExists
      ? businessCase.alternativeB.hosting.location
      : null,
    alternativeBHostingCloudServiceType: alternativeBExists
      ? businessCase.alternativeB.hosting.cloudServiceType
      : null,
    alternativeBHasUI: alternativeBExists
      ? businessCase.alternativeB.hasUserInterface
      : null,
    alternativeBPros: alternativeBExists
      ? businessCase.alternativeB.pros
      : null,
    alternativeBCons: alternativeBExists
      ? businessCase.alternativeB.cons
      : null,
    alternativeBCostSavings: alternativeBExists
      ? businessCase.alternativeB.costSavings
      : null,
    lifecycleCostLines
  };
};

export const hostingTypeMap: any = {
  cloud: 'Yes, in the cloud (AWS, Azure, etc.)',
  dataCenter: 'Yes, at a data center',
  none: 'No, hosting is not needed'
};
