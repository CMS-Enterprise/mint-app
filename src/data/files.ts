import { FileUploadForm, UploadedFile } from 'types/files';

export const fileUploadFormInitialData: FileUploadForm = {
  file: {} as File,
  filename: '',
  uploadURL: '',
  documentType: {
    commonType: null,
    otherType: ''
  }
};

export const fileUploadTableInitialData: UploadedFile = {
  filename: '',
  uploadURL: '',
  downloadURL: ''
};

export const prepareFileUploadForApi = (fileUpload: FileUploadForm): any => {
  const fileURL = {
    fileName: fileUpload?.file?.name,
    fileType: fileUpload?.file?.type,
    fileSize: fileUpload?.file?.size
  };

  return fileURL;
};

export const prepareFileUploadForApp = (fileUpload: any): FileUploadForm => {
  const upload = {
    file: fileUpload.file,
    filename: fileUpload.filename,
    uploadURL: fileUpload.URL,
    documentType: {
      commonType: null,
      otherType: ''
    }
  };
  return upload;
};

export const prepareUploadedFileForApp = (fileUpload: any): UploadedFile => {
  const uploadedFile = {
    file: fileUpload.file,
    filename: fileUpload.filename,
    uploadURL: fileUpload.uploadURL,
    downloadURL: ''
  };
  return uploadedFile;
};
