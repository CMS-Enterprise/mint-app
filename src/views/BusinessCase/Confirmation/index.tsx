import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useParams } from 'react-router-dom';
import { IconNavigateBefore } from '@trussworks/react-uswds';

import PageHeading from 'components/PageHeading';
import { ImproveEasiSurvey } from 'components/Survey';
import { BusinessCaseModel } from 'types/businessCase';

const Confirmation = ({
  businessCase
}: {
  businessCase: BusinessCaseModel;
}) => {
  const { businessCaseId } = useParams<{ businessCaseId: string }>();
  const { t } = useTranslation();

  return (
    <div className="grid-container margin-bottom-7">
      <div>
        <PageHeading>
          {t('businessCase:submission.confirmation.heading')}
        </PageHeading>
        <h2 className="margin-bottom-8 text-normal">
          {t('businessCase:submission.confirmation.subheading', {
            referenceId: businessCaseId
          })}
        </h2>
        <ImproveEasiSurvey />
        <div>
          <Link
            to={`/governance-task-list/${businessCase.systemIntakeId}`}
            className="display-flex"
          >
            <IconNavigateBefore className="margin-x-05" aria-hidden />
            {t('businessCase:submission.confirmation.taskListCta')}
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Confirmation;
