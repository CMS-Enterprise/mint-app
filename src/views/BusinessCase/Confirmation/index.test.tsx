import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { businessCaseInitialData } from 'data/businessCase';

import Confirmation from './index';

describe('The Business Case Confirmation page', () => {
  const testBusinessCase = {
    ...businessCaseInitialData,
    systemIntakeId: '12345'
  };
  it('renders without crashing', () => {
    shallow(
      <MemoryRouter>
        <Confirmation businessCase={testBusinessCase} />
      </MemoryRouter>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Confirmation businessCase={testBusinessCase} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
