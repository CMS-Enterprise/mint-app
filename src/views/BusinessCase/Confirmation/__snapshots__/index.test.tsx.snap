// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Business Case Confirmation page matches the snapshot 1`] = `
<div
  className="grid-container margin-bottom-7"
>
  <div>
    <h1
      aria-live="polite"
      className="easi-h1"
      tabIndex={-1}
    >
      Your Business Case has been submitted
    </h1>
    <h2
      className="margin-bottom-8 text-normal"
    >
      Your reference ID is 
    </h2>
    <p
      className="margin-top-4"
    >
      Help us improve EASi. 
      <a
        aria-label="Open EASi survey in a new tab"
        href="https://www.surveymonkey.com/r/JNYSMZP"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tell us what you think of this service (opens in a new tab)
      </a>
    </p>
    <div>
      <a
        className="display-flex"
        href="/governance-task-list/12345"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="usa-icon margin-x-05"
          focusable={false}
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
        Go back to Governance Task List
      </a>
    </div>
  </div>
</div>
`;
